File: app/src/main/java/com/edittextpicker/aliazaz/edittextpicker/MainActivity.java
package com.edittextpicker.aliazaz.edittextpicker;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    Button btnSubmit;
    com.edittextpicker.aliazaz.EditTextPicker txtBoxRange, txtBoxRangeMaskPat, txtBoxDefault, txtDate, txtPhone;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initializingComponents();
        settingListeners();

    }

    private void initializingComponents() {
        btnSubmit = findViewById(R.id.btnSubmit);
        txtBoxRange = findViewById(R.id.txtBoxRange);
        txtBoxRangeMaskPat = findViewById(R.id.txtBoxRangeMaskPat);
        txtBoxDefault = findViewById(R.id.txtBoxDefault);
        txtDate = findViewById(R.id.txtDate);
        txtPhone = findViewById(R.id.txtPhone);
    }

    private void settingListeners() {

        btnSubmit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (validateComponents()) {
                    Toast.makeText(MainActivity.this, "Successfully submitted!!", Toast.LENGTH_SHORT).show();
                    clearFields();
                }

            }
        });

    }

    private boolean validateComponents() {

        if (!txtBoxRange.isRangeTextValidate())
            return false;

        if (!txtBoxRangeMaskPat.isRangeTextValidate())
            return false;

        if (!txtBoxDefault.isTextEqualToPattern())
            return false;

        if (!txtDate.isEmptyTextBox())
            return false;

        return txtPhone.isEmptyTextBox();
    }

    private void clearFields() {

        txtBoxRange.setText(null);
        txtBoxRangeMaskPat.setText(null);
        txtBoxDefault.setText(null);
        txtDate.setText(null);
        txtPhone.setText(null);

    }
}

File: app/src/main/java/com/edittextpicker/aliazaz/edittextpicker/MainActivity.kt
package com.edittextpicker.aliazaz.edittextpicker

import android.text.InputType
import android.widget.Toast
import com.edittextpicker.aliazaz.EditTextPicker
import com.edittextpicker.aliazaz.edittextpicker.databinding.ActivityMainBinding
import com.edittextpicker.aliazaz.repository.EditTextPickerBuilder

/*
* @author Ali Azaz Alam
* */
class MainActivity : BaseActivity<ActivityMainBinding>() {

    private var txtPicker: EditTextPicker? = null
    override fun getLayout(): Int = R.layout.activity_main

    override fun init() {
        settingListeners()
    }

    private fun settingListeners() {
        binding.btnSubmit.setOnClickListener {
            if (validateComponents()) {
                Toast.makeText(
                    this@MainActivity,
                    getString(R.string.form_submitted),
                    Toast.LENGTH_SHORT
                )
                    .show()
                clearFields()
            }
        }

        /*
        * Setting date mask to txtDate
        * */
        binding.txtDate.setMask("##-##-####").setRequired(false)

        binding.txtPhone.setMaskText("3453323212")

        // Create EditTextPicker programmatically
        txtPicker = EditTextPicker(this, EditTextPickerBuilder().apply {
            setRequired(true)
            setRangeValues(0.5f, 40.0f)
            setMask("##.##")
            setPattern("^(\\d{2,2}\\.\\d{2,2})$")
        }.build()).apply {
            hint = "##.##"
            inputType = InputType.TYPE_CLASS_NUMBER
        }
        binding.llLayout.addView(txtPicker)

    }

    private fun validateComponents(): Boolean {
        if (!binding.txtBoxRange.isRangeTextValidate()) return false
        else if (txtPicker?.isRangeTextValidate() == false) return false
        else if (!binding.txtBoxDefault.isTextEqualToPattern()) return false
        else if (!binding.txtDate.isEmptyTextBox()) return false
        return binding.txtPhone.isEmptyTextBox()
    }

    private fun clearFields() {
        binding.txtBoxRange.text = null
        txtPicker?.text?.clear()
        binding.txtBoxDefault.text = null
        binding.txtDate.text = null
        binding.txtPhone.text = null
    }
}